cpu/sam0_common/periph/adc.c

```diff
@@ -120,33 +134,73 @@ static int _adc_configure(adc_res_t res)
     /* Power on */
 #ifdef CPU_SAML1X
     MCLK->APBCMASK.reg |= MCLK_APBCMASK_ADC;
+#endif
+#ifdef MCLK_APBDMASK_ADC0
+    if (ADC_DEV == ADC0) {
+        MCLK->APBDMASK.reg |= MCLK_APBDMASK_ADC0;
+    } else {
+        MCLK->APBDMASK.reg |= MCLK_APBDMASK_ADC1;
+    }
 #else
     MCLK->APBDMASK.reg |= MCLK_APBDMASK_ADC;
 #endif
+
+#ifdef ADC0_GCLK_ID
     /* GCLK Setup */
-    GCLK->PCHCTRL[ADC_GCLK_ID].reg = GCLK_PCHCTRL_CHEN | GCLK_PCHCTRL_GEN_GCLK0;
-    /* Set Voltage Reference */
-    ADC->REFCTRL.reg = ADC_REF_DEFAULT;
+    if (ADC_DEV == ADC0) {
+        GCLK->PCHCTRL[ADC0_GCLK_ID].reg = GCLK_PCHCTRL_CHEN |
+            GCLK_PCHCTRL_GEN(ADC_GCLK_SRC);
```

Comment:
```suggestion
        GCLK->PCHCTRL[ADC0_GCLK_ID].reg = GCLK_PCHCTRL_CHEN
                                        | GCLK_PCHCTRL_GEN(ADC_GCLK_SRC);
```

looks a bit nicer imho

Reply:
The bitwise OR was already in the line before the line break in some other parts of the driver. 

To get the bit operations directly one below the other is unfortunately only possible in 2-3 places.I tried to keep it consistent with the rest of the code.

But I've moved the bitwise or into the next line where it made sense for me now. If it still doesn't quite fit, just let me know.